src/shelly_sw_service.c

```diff
@@ -100,6 +124,16 @@ static void shelly_sw_set_state_ctx(struct shelly_sw_service_ctx *ctx,
       mgos_set_timer(600, 0, do_reset, ctx);
     }
   }
+
+  handle_auto_off(ctx, source, new_state);
+}
+
+static void do_auto_off(void *arg) {
+  s_auto_off_timer_id = MGOS_INVALID_TIMER_ID;
+  struct shelly_sw_service_ctx *ctx = arg;
+  const struct mgos_config_sw *cfg = ctx->cfg;
+  if (cfg->auto_off)  // Don't set state if auto off has been disabled during timer run
```

Comment:
please add braces here, it's a requirement for multiline conds for maximum clarity.

Reply:
it's just a comment, but sure, I've added brackets